### ISSUE TITLE
column formatting formatting for each column in table visual 

### ISSUE THREAD

**Issue by akash273** [Microsoft/PowerBI-visuals#228]
Hi All,
 
I have developed custom table visual in power bi .  Now my requirement is about conditional column formatting.
 
 
My table visual i made to work conditional formatting for 3rd column based my indicators but know , Now my requirement is i have 5 more columns with their indicators related to each one of them. Is there any way to get the columns in formatting pane so that i can color them according to thier indicators by selecting each column.

![table pic](https://user-images.githubusercontent.com/23463341/28643995-b400708c-7225-11e7-9d79-e950b4876f85.PNG)

**Comment by ABKimmel** [comment]
Hi akash,

Are you asking how you would add options to the formatting pane, so that you can set the color of a column?

**Comment by akash273** [comment]
Yes, I wanted to color the bars according to their indicators such as 1-green,2-yellow,3-red . As of now i hided and hard coded the indicator column in backend to not show that but have to color according to that. So similarly now my requirement has more indicators columns for each column now ,now i need to select like column according to different indicators column i should be able to do it ,IS that possible.

**Comment by ABKimmel** [comment]
Let me get this straight:

You were originally required to color your data based on a set of colors, such as 1 should be green, 2 should be yellow, and 3 should red. Now you need to color your data based on the value of a column or columns?

**Comment by akash273** [comment]
![capture](https://user-images.githubusercontent.com/23463341/28674486-8935226a-72b3-11e7-92cf-f28290ba79d6.PNG)

**Comment by ABKimmel** [comment]
I'm afraid this doesn't really help me understand your problem.

**Comment by akash273** [comment]
According to the Sales indicator column the sales% column bars colors  should be changed.


Did u get it now

**Comment by ABKimmel** [comment]
Ahh, so you want to define a setting that let's you pick the color that corresponds to the Sales indicator? And the possible indicators are all listed in the Sales Indicator column?

If that's the case, I would suggest creating a list of the unique values of Sales Indicator, then enumerating a settings object for it. I cover something like this [here](https://github.com/ABKimmel/Power-BI-Visualization-Development-Tutorial/blob/master/docs/advanced/SettingsDefinitions.md). You probably want to do something like the `categoryColors` example.

**Comment by akash273** [comment]
Thanks for the reply,
In my requirement i will be having more indicators columns like sales

Is it possible to keep all indicators columns together so that for each column now i can select which indicator it should have.

**Comment by ABKimmel** [comment]
You could probably repeat the process of enumerating the indicators for each column in a different setting. Or if you're asking how to set which kind of indicator you are assigning to a column (Sales indicator, revenue indicator, orders indicator, etc.), you could create a separate setting for that using the `Enumeration` type and specifying the column names.

**Comment by akash273** [comment]
okay thank you for your help

I will do that done and comment you again if any issues please reply me..


Thanks
Akash ragi

**Comment by akash273** [comment]
Hi Kimmel
Now, the issue is there are a few columns say, Sales, Revenue etc. For each of those columns I have another column which would define its indicator. For example, for sales column I have sales indicator column and for revenue column there is a revenue indicator column. I should map each row data of that column with another row data of indicator column. So how would i bind these indicators to that particular data of the original column. and then create color formatting for it.
